app/src/renderer/vuex/modules/send.js

```diff
@@ -179,12 +194,36 @@ export default ({ node }) => {
 
       // broadcast transaction with signatures included
       const signedTx = state.externals.createSignedTx(tx, signature)
-      const body = state.externals.createBroadcastBody(signedTx)
-      const res = await node.postTx(body).catch(handleSDKError)
+      const body = state.externals.createBroadcastBody(signedTx, `sync`)
+      const res = await state.node.postTx(body).catch(handleSDKError)
 
       // check response code
       assertOk(res)
-      commit(`setNonce`, String(parseInt(state.nonce) + 1))
+
+      // sync success
+      if (res.height !== `0`) {
+        commit(`setNonce`, String(parseInt(state.nonce) + 1))
+        return
+      }
+
+      return dispatch(`queryTxInclusion`, res.txhash)
+    },
+    // wait for inclusion of a tx in a block
+    async queryTxInclusion({ state }, txHash) {
+      let success = false
+      let iterations = state.txQueryIterations // 30 * 2s = 60s max waiting time
+      while (!success && iterations-- > 0) {
+        await new Promise(resolve => setTimeout(resolve, state.txQueryTimeout))
+        try {
+          await state.node.tx(txHash)
+          success = true
```

Comment:
I would try to simplify a by removing the variable `success` and replacing it here with a `break` + removing the `!success` and line `213`

Reply:
and moving the await inside the catch, so the very first time it won't wait (one failure extra) but it looks as a trade-off to improve a bit the easiness to read this loop

Reply:
nice ideas